static-build/CMakeLists.txt

```diff
@@ -290,7 +220,6 @@ ExternalProject_Add(tarantool
         -DCMAKE_INSTALL_LOCALSTATEDIR=<INSTALL_DIR>/var
         -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
         -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
-        -DCMAKE_FIND_USE_CMAKE_SYSTEM_PATH=FALSE
```

Comment:
Wow, that's feels dangerous. It was added to ensure that we don't pull any external dependencies into the executable (dynamic linking)[^1]. However there is `static-build/test/CheckDependencies.cmake`, which should check it after the build. However (2), how it is passed if you added libunwind here, whether it works at all?

@CuriousGeorgiy It looks strange and I would appreciate if you'll add a bullet into #6998 to don't forget to look at this again and discuss.

[^1]: It works only on cmake >= 3.16 and no-op on older cmake versions, but it is okay.

Reply:
The test works fine:
```
$ ctest -V -R check-dependencies
[...]
1: CMake Error at cmake/CheckDependencies.cmake:43 (message):
1:   Blocklisted dependency: /lib/x86_64-linux-gnu/libggg.so
```

Why it should fail for `libunwind`, which is statically linked?

Reply:
Sure, if it is liked statically, the test shouldn't see the library. I guess that I missed this point when I wrote the original message.

Reply:
macOS static build requires `-DCMAKE_FIND_USE_CMAKE_SYSTEM_PATH=TRUE`, see this thread:
- https://github.com/tarantool/tarantool/pull/9084#discussion_r1313400540

Reply:
/cc @Totktonada

Reply:
Since it is necessary: OK, let it be. At least, we have a test on the topic.